app/views/questionnaire.py

```diff
@@ -238,10 +235,8 @@ def submit_answers(eq_id, form_type, collection_id, metadata, answer_store):
         submitter = SubmitterFactory.get_submitter()
         message = convert_answers(metadata, g.schema_json, answer_store, path_finder.get_routing_path())
         submitter.send_answers(message)
-        questionnaire_store = get_questionnaire_store(current_user.user_id, current_user.user_ik)
-        metadata['submitted'] = True
-        questionnaire_store.metadata = metadata
-
+        session_storage.store_survey_completed_metadata(metadata['tx_id'], metadata['period_str'], metadata['ru_ref'])
+        get_questionnaire_store(current_user.user_id, current_user.user_ik).delete()
```

Comment:
as `get_thank_you` no longer has the `login_required` decorator i think @collisdigital is correct